### PR TITLE
add support for pathlib to load and dump

### PR DESCRIPTION
**Pull request by karlind**

### CODE REVIEW

**Comment by karlind** [comment]
#3

File: mmcv/fileio/io.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from pathlib import Path
```
Comment (hellock) [inline mmcv/fileio/io.py]: Add a blank line between system imports and relative imports.

File: mmcv/fileio/io.py [hunk 2]
```diff
@@ -16,7 +17,7 @@ def load(file, file_format=None, **kwargs):
     This method provides a unified api for loading data from serialized files.
 
     Args:
-        file (str or file-like object): Filename or a file-like object.
+        file (str or Path or file-like object): Filename or a file-like object.
```
Comment (hellock) [inline mmcv/fileio/io.py]: We can use :obj:\`Path\` instead of Path for better visualization.

**Comment by openmmlab-v** [comment]
Hi @karlind ！First of all, we want to express our gratitude for your significant PR in the mmcv project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**